src/message.js

```diff
@@ -140,41 +141,56 @@ Message.prototype.decrypt = async function(privateKey, sessionKey, password) {
 };
 
 /**
- * Decrypt an encrypted session key either with a private key or a password.
- * @param  {Key} privateKey    (optional) private key with decrypted secret data
- * @param  {String} password   (optional) password used to decrypt
+ * Decrypt an encrypted session key either with private keys or passwords.
+ * @param  {Array<Key>} privateKeys    (optional) private key with decrypted secret data
+ * @param  {Array<String>} passwords   (optional) password used to decrypt
  * @return {Array<{ data:Uint8Array, algorithm:String }>} array of object with potential sessionKey, algorithm pairs
  */
-Message.prototype.decryptSessionKeys = function(privateKey, password) {
+Message.prototype.decryptSessionKeys = function(privateKeys, passwords) {
   var keyPackets = [];
   return Promise.resolve().then(async () => {
-    if (password) {
+    if (passwords) {
       var symESKeyPacketlist = this.packets.filterByTag(enums.packet.symEncryptedSessionKey);
       if (!symESKeyPacketlist) {
         throw new Error('No symmetrically encrypted session key packet found.');
       }
       await Promise.all(symESKeyPacketlist.map(async function(packet) {
-        try {
-          await packet.decrypt(password);
-          keyPackets.push(packet);
-        } catch (err) {}
+        for (var i = 0; i < passwords.length; i++) {
+          try {
+            // eslint-disable-next-line no-await-in-loop
+            await packet.decrypt(passwords[i]);
+            keyPackets.push(packet);
+          } catch (err) {}
+        }
       }));
 
-    } else if (privateKey) {
+    } else if (privateKeys) {
       var pkESKeyPacketlist = this.packets.filterByTag(enums.packet.publicKeyEncryptedSessionKey);
       if (!pkESKeyPacketlist) {
         throw new Error('No public key encrypted session key packet found.');
       }
-      var privateKeyPacket = privateKey.getKeyPacket(this.getEncryptionKeyIds());
-      if (!privateKeyPacket.isDecrypted) {
-        throw new Error('Private key is not decrypted.');
-      }
       await Promise.all(pkESKeyPacketlist.map(async function(packet) {
-        if (packet.publicKeyId.equals(privateKeyPacket.getKeyId())) {
-          try {
-            await packet.decrypt(privateKeyPacket);
-            keyPackets.push(packet);
-          } catch (err) {}
+        for (var i = 0; i < privateKeys.length; i++){
+          var privateKeyPackets;
+          if (packet.publicKeyId.isWildcard()) {
```

Comment:
What about making getKeyPacket return AllKeyPackets if the public key ID is zero?  Then you can eliminate the if/else here.

Another thing to think about is to call this.getEncryptionKeyIds, then loop over privateKeys calling getKeyPacket(), then construct an object with properties keyed by keyID merging the results, and only then looping over privateKeyPackets. That would eliminate the nested loops you have here and probably be more efficient. Let me know what you think.